File: src/client/cell.cpp
#include "cell.h"

using namespace std;

Cell::Cell(int x, int y) : point_(Point(x, y)) {
  player_ = NULL;
}

Cell::Cell(Point p) : point_(p) {
  player_ = NULL;
}

Cell::~Cell() {

}

Player *Cell::getPlayer() {
  return player_;
}

char Cell::getSymbol() const {
  if (player_->getSymbol()) {
    return player_->getSymbol();
  }
  return ' ';
}

bool Cell::isEmpty() {
  return player_ == NULL;
}

void Cell::fill(Player *p) {
  player_ = p;
}

int Cell::getX() const {
  return point_.getX();
}

int Cell::getY() const {
  return point_.getY();
}

void Cell::setX(int x) {
  point_.setX(x);
}

void Cell::setY(int y) {
  point_.setY(y);
}

bool Cell::operator==(const Cell &otherCell) const {
  return getX() == otherCell.getX() && getY() == otherCell.getY();
}

bool Cell::operator!=(const Cell &otherCell) const {
  return point_.getX() != otherCell.getX() || point_.getY() != otherCell.getY();
}

std::ostream &operator<<(std::ostream &out, Cell const &c) {
  return out << "(" << c.getX() + 1 << ", " << c.getY() + 1 << ")";
}

File: Cell.java
public class Cell {
    private Point point;
    private Player player;

    /**
     * Constructor for a cell by x, y coordinates.
     * @param x coordinate for the cell.
     * @param y coordinate for the cell.
     */
    public Cell(int x, int y) {
        this.point = new Point(x, y);
    }

    /**
     * Constructor for a cell by point.
     * @param point coordinates by point.
     */
    public Cell(Point point) {
        this.point = point;
    }

    /**
     * Returns the occupying player's symbol.
     * @return occupying player's symbol.
     */
    public char getSymbol() {
        return this.player == null ? ' ' : this.player.getSymbol();
    }

    /**
     * Returns the x coordinate of the cell.
     * @return x coordinate of the cell.
     */
    public int getX() {
        return this.point.getX();
    }

    /**
     * Sets the x coordinate of the cell to given x.
     * @param x new coordinate of the cell.
     */
    public void setX(int x) {
        this.point.setX(x);
    }

    /**
     * Returns the y coordinate of the cell.
     * @return y coordinate of the cell.
     */
    public int getY() {
        return this.point.getY();
    }

    /**
     * Sets the y coordinate of the cell to given y.
     * @param y new coordinate of the cell.
     */
    public void setY(int y) {
        this.point.setY(y);
    }

    /**
     * Returns the occupying player of the cell.
     * @return occupying player of the cell.
     */
    public Player getPlayer() {
        return this.player;
    }

    /**
     * Checks if the cell is empty.
     * @return true if cell is empty, false otherwise.
     */
    public boolean isEmpty() {
        return this.player == null;
    }

    /**
     * Fills the cell with a player.
     * @param player player to fill the cell with.
     */
    public void fill(Player player) {
        this.player = player;
    }
}